### ISSUE TITLE
LDAP Result Code 32 "No Such Object"

### ISSUE THREAD

**Issue by funkel1989** [authelia/authelia#2652]
Authelia is reporting an LDAP Result Code no such object found against my activedirectory domain. I've looked at and validated my config is right against the other issue opened with this same thing but it doesn't seem to work. 

My Config
```yaml
ldap:
      implementation: activedirectory
      url: ldap://<ldapIPAddress>
      timeout: 5s
      start_tls: false
      tls:
        server_name: ""
        skip_verify: true
        minimum_version: TLS1.2
      base_dn: dc=<domain>,dc=com
      username_attribute: sAMAccountName
      #additional_users_dn: ou=Users
      #users_filter: (&({username_attribute}={input})({mail_attribute}={input})(sAMAccountType=805306368)(!(userAccountControl:1.2.840.113556.1.4.803:=2))(!(pwdLastSet=0))(objectCategory=person)(objectClass=user))    << this is what I'd like to use but even the most simple filter doesn't work
      users_filter: (&({username_attribute}={input})(objectClass=user))
      #additional_groups_dn: ou=Groups
      groups_filter: (&(objectCategory=group)(member={dn}))
      group_name_attribute: cn
      mail_attribute: mail
      display_name_attribute: displayName
      user: <domain>\<user>    ##  <<< I had to format the username like this for it to work which I thought was odd.
```

Error:
```
error while checking password for user funkel: cannot find user DN of user '<user>'. Cause: LDAP Result Code 32 \"No Such Object\": 0000208D: NameErr: DSID-0310028B, problem 2001 (NO_OBJECT), data 0, best match of:\n\t'DC=<domain>,DC=com'\n\x00" method=POST path=/api/firstfactor remote_ip=10.42.1.0 stack="github.com/authelia/authelia/v4/internal/middlewares/authelia_context.go:66 (*AutheliaCtx).Error\ngithub.com/authelia/authelia/v4/internal/handlers/response.go:150 handleAuthenticationUnauthorized\ngithub.com/authelia/authelia/v4/internal/handlers/handler_firstfactor.go:102 FirstFactorPost.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/authelia_context.go:51 AutheliaMiddleware.func1.1\ngithub.com/fasthttp/router@v1.4.3/router.go:414 (*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14 LogRequestMiddleware.func1\ngithub.com/valyala/fasthttp@v1.30.0/server.go:2266 (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.30.0/workerpool.go:223 (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.30.0/workerpool.go:195 (*workerPool).getCh.func1\nruntime/asm_amd64.s:1581 goexit"
```

when I do the following

```powershell
Get-ADObject -LDAPFilter "(&(objectCategory=user)(sAMAccountName=<user>))" -SearchBase "DC=mycavanaughnetwork,DC=com"
```
I am successfully returned the user that I expect to return without an issue.

What am I doing wrong here?

**Comment by james-d-elliott** [comment]
You need to add the active directory implementation.  Though that should not affect it. Your user attribute should be the distinguishedName of the user used to bind, what was it before?

```yaml
authentication_backend:
  ldap:
    implementation: activedirectory

**Comment by funkel1989** [comment]
> You need to add the active directory implementation. Though that should not affect it. Your user attribute should be the distinguished name of the user used to bind, what was it before?
> 
> ```yaml
> authentication_backend:
>   ldap:
>     implementation: activedirectory
> ```

Sorry I had accidentally deleted that line from my config that i posted when I was removing a bunch of comments but I do already have that there.

**Comment by funkel1989** [comment]
ok we can close this. I had two issues. the user I created for this didn't seem to have the right permissions to query ldap even though it was in the administrators group because when I switched to the administrator user I was able to query users. After that I learned my group filter was wrong and I switched to using the group filter suggested in your documentation which seems to have resolved my issues. Sorry for the noise!
